plutus-core/plutus-core/src/PlutusCore/Crypto/BLS12_381/G1.hs

```diff
@@ -93,9 +98,8 @@ compress (Element a) = BlstBindings.blsCompress @BlstBindings.Curve1 a
      * The bytestring does represent a point on the E1 curve, but the
        point is not in the G1 subgroup.
 -}
--- TODO: perhaps make this emit the error in the Left case.
```

Comment:
> should convert from `Either` to `Emitter` Any opinions?

I have a strong opinion: SECP functions are screwed up, because we didn't give people the right tools and they had no options, but to write that screwed up thing.

I see four sensible options:

1. keep what we have, remove all explicit emitting and automatically emit error messages
2. add (well, return) `MonadEmitter` and allow people to use `MonadError` + `MonadEmitter`. This is nice, but might be inefficient.
3. add `MonadEmitter` and provide an instance for `ExceptT` and ask people to use `ExceptT` + `Emitter` as a concrete monad stack
4. add `emitMakeKnownM` (or turn `emit` into the method of an added `MonadEmitter`) and ask people to use `MakeKnownM`, which is effectively `MonadError Text` + `MonadEmitter`. This is not nice and limits your error type to be `Text`, but it's efficient.
5. add a version of `MakeKnownM` working with any error type and ask people to use that. Nicer than the previous option and is still efficient, but more boilerplate. 


Of these the first one by far makes most sense to me and I hate the fact that `EvaluationResult` et al don't have an actual damn error message to them. There's literally zero reason to optimize for them not having an error message, given that they terminate evaluation anyway.

The next most sensible option seems to be 3 to me. I haven't played around with it, though, maybe it's trickier than it seems to me.

The current alignment definitely needs to be fixed, it's pretty much inviting bugs (`EvaluationFailure` stops execution, but `EvaluationFailure` wrapped in `Emitter` doesn't if you keep evaluating things in `Emitter`).

Reply:
> I have a strong opinion

OK, we're not going to do any of that here! I'll leave it as it is and we can make a ticket to implement a general fix for this at some point.

Reply:
> OK, we're not going to do any of that here!

Definitely. 

> I'll leave it as it is and we can make a ticket to implement a general fix for this at some point.

[Done](https://input-output.atlassian.net/browse/PLT-6383).